pinot-common/src/main/java/org/apache/pinot/common/metrics/BrokerQueryPhase.java

```diff
@@ -30,6 +30,8 @@
   QUERY_EXECUTION,
   QUERY_ROUTING,
   SCATTER_GATHER,
+  SERVER_PROCESSING,
```

Comment:
Why do we need to emit a server metric from broker ? Each server is already emitting its own totalQueryTime so I don't think we need this. Will lead to more confusion IMO.

Reply:
The primary aim for this PR is to extract the time spent on netty from the time spent on scatter_gather. We should have a metric to actually measure the network time. Now that the broker query phase has all the phases from end to end, it'd be good to have the server total query time here instead of leaving it to the server metrics.

Reply:
We have often seen the case when either server or broker goes through a gc just as the response is being sent by server or received by broker. In this case, gc time will show up as  the "network time" you are measuring? On the server side, it will be the gc time included as "server time", but on the broker, if it receives the bytes from server and then slips into gc (presumably to copy the bytes in, so it is likely to happen), then we will see the gc time as netty time, right?

What exactly is the problem faced that we could not get an answer for using the current set of metrics?

Reply:
We couldn't avoid gc from happening. Wherever the gc happens is the place we should account for the time. I've just refactored the code in the latest push, which makes it more clear. Basically the scatter & gather consists of 3 parts: 
* request sent from broker to server
* query being processed (totalQueryTime)
* response sent from server to broker

If gc happens on any of the steps above, it should be measured in just that step.